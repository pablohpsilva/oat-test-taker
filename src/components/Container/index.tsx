import styled from 'styled-components/macro'

export default styled.div`
    max-width: var(--dim-site-width);
    padding: 0 var(--dim-site-offset);
`