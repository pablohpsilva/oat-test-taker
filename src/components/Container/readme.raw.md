# Container

This is a container component responsible for making its
`children` always on the screen.

## Props

| Property | Description | Type | Default |
| -------- | ----------- | ---- | ------- |
| `-`      | -           | `-`  | `-`     |
