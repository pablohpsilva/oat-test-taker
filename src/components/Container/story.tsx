import React from 'react'
import { storiesOf } from '@storybook/react'

import { Container } from '@/components'
import readme from './readme.raw.md'

const ContainerComp = () => <Container>I'm centered!</Container>

const opts = {
    notes: { markdown: readme }
}

storiesOf('Components|Container', module)
    .addDecorator(storyFn => <div>{storyFn()}</div>)
    .add('Example', () => <ContainerComp />, opts)
