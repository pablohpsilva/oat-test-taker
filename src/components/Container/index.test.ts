import React from 'react'
import Container from './'

test('should render the Container component', async () => {
    const wrapper = shallow(<Container />)

    expect(wrapper.first().hasClass('Container')).toMatchSnapshot()
    expect(wrapper).toMatchSnapshot()
})
