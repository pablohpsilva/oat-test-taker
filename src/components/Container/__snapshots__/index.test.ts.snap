// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Container component 1`] = `false`;

exports[`should render the Container component 2`] = `
.c0 {
  max-width: var(--dim-site-width);
  padding: 0 var(--dim-site-offset);
  width: 100%;
  position: relative;
}

@media screen and (max-width:1200px) {
  .c0 {
    padding: 0;
  }
}

<div
  className="c0"
/>
`;
